docs/content/guides/dagster-pipes/integrating-subprocess-with-dagster-pipes.mdx

```diff
@@ -1,6 +1,244 @@
+- [Stream structured metadata from subprocess back to Dagster](/guides/dagster-pipes/modifying-existing-code-to-work-with-dagster-pipes#send-structured-metadata-to-dagster)
+
+---
+
+## Step 3: Run the subprocess from the Dagster UI
+
+After completing Step 2, you should have created a standalone Python file called `external_code.py` which looks like the following:
+
+```python file=/guides/dagster/dagster_pipes/subprocess/external_code_finished.py
+import pandas as pd
+from dagster_pipes import PipesContext, open_dagster_pipes
+
+
+def main():
+    orders_df = pd.DataFrame({"order_id": [1, 2], "item_id": [432, 878]})
+    total_orders = len(orders_df)
+    # get the Dagster Pipes context
+    context = PipesContext.get()
+    # send structured metadata back to Dagster
+    context.report_asset_materialization(metadata={"total_orders": total_orders})
+    # report data quality check result back to Dagster
+    context.report_asset_check(
+        asset_key="my_asset",
+        passed=orders_df[["item_id"]].notnull().all().bool(),
+        check_name="no_empty_order_check",
+    )
+
+
+if __name__ == "__main__":
+    # connect to Dagster Pipes
+    with open_dagster_pipes() as context:
+        main()
+```
+
+In this step, you’ll execute the subprocess you created in Step 1.2 from the Dagster UI.
+
+1. In a new command line session, run the following to start the UI:
+
+```bash
+dagster dev
+```
+
+2. Navigate to localhost:3000, where you should see the UI:
+
+<Image
+alt="Asset in the UI"
+src="/images/guides/dagster-pipes/subprocess-1-asset.png"
+width={1264}
+height={714}
+/>
+
+3. Click **Materialize** located in the top right to run your code:
+
+<Image
+alt="Materialize asset"
+src="/images/guides/dagster-pipes/subprocess-2-materialize.png"
+width={1326}
+height={744}
+/>
+
+4. Navigate to the [Run details](/concepts/webserver/ui#run-details) page, where you should see the structured asset metadata and asset check result:
+
+<Image
+alt="Events in the run details page"
+src="/images/guides/dagster-pipes/subprocess-3-events.png"
+width={1272}
+height={718}
+/>
+
+---
+
+## Examples
+
+### Specify environment variables and extras to a subprocess
+
+The `run` method to the `PipesSubprocessClient` resource also accepts `env` and `extras` , which allow you to specify environment variables and extra arguments when executing the subprocess:
+
+```python file=/guides/dagster/dagster_pipes/subprocess/dagster_subprocess_env_extras.py
+import os
+import shutil
+
+from dagster import (
+    AssetExecutionContext,
+    PipesSubprocessClient,
+    asset,
+    file_relative_path,
+)
+
+
+@asset
+def subprocess_asset(
+    context: AssetExecutionContext, pipes_subprocess_client: PipesSubprocessClient
+):
+    cmd = [shutil.which("python"), file_relative_path(__file__, "external_code.py")]
+    return pipes_subprocess_client.run(
+        command=cmd,
+        context=context,
+        extras={"foo": "bar"},
```

Comment:
IMO this only makes sense if you show the external code access extras, otherwise I have no idea how this is used